### PR TITLE
Join count filter criteria

### PR DESCRIPTION
**Pull request by WithoutPants**
Extension of #1234 

* Adds `tag_count` criterion and sorting option to Performer, Scene, Gallery and Image filters.
* Adds `performer_count` criterion and sorting option to Scene, Gallery and Image filters.
* Adds `scene_count`, `image_count`, `gallery_count` criteria to Performer filter.

### CODE REVIEW

File: graphql/schema/types/filters.graphql [hunk 1]
```diff
@@ -61,6 +61,14 @@ input PerformerFilterType {
...
   """Filter to only include performers with these tags"""
   tags: MultiCriterionInput
+  """Filter by tag count"""
+  tag_count: IntCriterionInput
+  """Filter by scene count"""
+  scene_count: IntCriterionInput
+  """Filter by scene count"""
+  image_count: IntCriterionInput
+  """Filter by scene count"""
+  gallery_count: IntCriterionInput
```
Comment (peolic) [inline graphql/schema/types/filters.graphql:71]:
```suggestion
  """Filter by image count"""
  image_count: IntCriterionInput
  """Filter by gallery count"""
  gallery_count: IntCriterionInput
```

**Review by peolic — CHANGES_REQUESTED** [review]
Seems to work well (have only tested scenes & performers (+ performer tags), no gallery or images)

File: pkg/sqlite/sql.go [hunk 2]
```diff
@@ -218,6 +223,11 @@ func getMultiCriterionClause(primaryTable, foreignTable, joinTable, primaryFK, f
 	return whereClause, havingClause
 }
 
+func getCountCriterionClause(primaryTable, joinTable, primaryFK string, criterion models.IntCriterionInput) (string, int) {
+	lhs := fmt.Sprintf("(SELECT COUNT(*) FROM %s s WHERE s.%s = %s.id)", joinTable, primaryFK, primaryTable)
```
Comment (mrbrdo) [inline pkg/sqlite/sql.go:227]: I believe you are missing the `IFNULL(..., 0)` around the query, as I had originally. Without that I believe it won't work properly for count = 0 condition.

**Review by mrbrdo — CHANGES_REQUESTED** [review]
I believe that filter query needs the IFNULL addition.

**Comment by bnkai** [comment]
- the tag_count is missing from the UI's sort order in the performers page. (manually editing the URL and using it seems to work ok)
- the IS NULL / IS NOT NULL doesnt make sense/work for the _count filters as mentioned in the above post. Is Null returns nothing while Is not Null returns all entries. 
- you can probably ignore this. The  `/scenes?sortby=tag_count&sortdir=desc` was giving me 4 irrelevant scenes as first with fewer tags. I edited->saved those scenes and the sort order is fine now. Not sure why that happened.

**Review by bnkai — APPROVED** [review]

**Pull request merged**